camel-k-loader-groovy/src/main/groovy/org/apache/camel/k/loader/groovy/dsl/ComponentsConfiguration.groovy

```diff
@@ -48,10 +49,10 @@ class ComponentsConfiguration {
         }
 
         if (type.isAssignableFrom(component.class)) {
-            callable.resolveStrategy = Closure.DELEGATE_FIRST
+            // Just make sure the closure context is belong to component
+            callable.resolveStrategy = Closure.DELEGATE_ONLY
```

Comment:
`Closure.DELEGATE_FIRST` was done on purpose to be able to reuse methods on the `Component` class (although there's no method right now)

Reply:
If I set the Closure.DELEGATE_FIRST, I cannot throw the exception from propertyMissing method of ComponentsConfiguration.groovy. Is there any other way to address this issue?

Reply:
PropertyBindingException is propagated

Reply:
I tried to throw the PropertyBindingException from the propertyMissing method, I cannot get the exception when set the Closure.DELEGATE_FIRST.  Could you try to run code yourself?
It looks like Closure.DELEGATE_FIRST just eat up all the exception which is thrown from  ComponentsConfiguration.propertyMissing method.